terraform/modules/data_share_processor/data_share_processor.tf

```diff
@@ -228,6 +232,29 @@ POLICY
   }
 }
 
+# Besides the validation bucket owned by the peer data share processor, we write
+# validation batches into a bucket we control so that we can be certain they
+# be available when we perform the aggregation step.
+resource "google_storage_bucket" "own_validation_bucket" {
+  provider = google-beta
+  name     = "${local.resource_prefix}-own-validation"
+  location = var.gcp_region
+  # Force deletion of bucket contents on bucket destroy. Bucket contents would
+  # be re-created by a subsequent deploy so no reason to keep them around.
+  force_destroy               = true
+  uniform_bucket_level_access = true
+}
+
+# Permit the workflow manager and facilitator service account to manage the
+# bucket
+resource "google_storage_bucket_iam_binding" "own_validation_bucket_admin" {
+  bucket = google_storage_bucket.own_validation_bucket.name
+  role   = "roles/storage.objectAdmin"
```

Comment:
Out of curiosity does it need admin level permissions - or would objectCreator be enough?

Reply:
Excellent question, I was a little surprised at this, too. Per [GCP docs](https://cloud.google.com/storage/docs/access-control/iam-roles), `storage.ObjectAdmin` gets you `storage.objects.*`. I thought that meant `storage.objects.create`, `storage.objects.get`, `storage.objects.delete` and `storage.objects.list`, all of which are needed to read, write and delete objects. Now that I look at it again, it _also_ must get you `storage.objects.getIamPolicy` and `storage.objects.setIamPolicy`, which we _don't_ want to grant to this service account. Annoyingly, I don't see a predefined role that includes `storage.objects.delete`, so I think I need to create a custom role here that include that permission.

Reply:
turns out `storage.legacyBucketWriter` fits the bill, and is fine to use in spite of the ominous `legacy`.